src/pose_estimating.h

```diff
@@ -0,0 +1,229 @@
+   */
+  float upper_angle;
+  ConnectedJoint * lower_connected_joint;
+  /* 
+   * @brief Angle(in degrees) of lower joint measured anticlockwise from bottom left corner  
+   */
+  float lower_angle;
+}
+
+/* 
+ *  @brief The `Pose::Pose` representation of a human's pose, containing all the expected `Pose::ConnectedJoint 
+ */
+struct Pose {
+  ConnectedJoint head;
+  ConnectedJoint neck;
+  ConnectedJoint shoulder;
+  ConnectedJoint hip;
+  ConnectedJoint knee;
+  ConnectedJoint foot;
+}
+
+/*
+ * @brief This class handles representations of the user's pose and calculates any updates to their pose that is required.
+ * 
+ * This class stores representations of the users current pose and their calibrated ideal pose, from the results of the PostProcessor class.
+ *
+ * This class handles:
+ * Updating current user's pose or ideal user's pose
+ * Calculating the angle between ConnectedJoints
+ * Comparing the current user's pose and ideal user's pose
+ *
+ * General use should just consist of updateAndCheckCurrentPose()
+ * 
+ */
+class PoseEstimater {
+ private:
+  /*
+   * @brief Calculates the angle(in degrees) between two points, anticlockwise from origin / bottom left corner of image
+   * @param x1 Relative X co-ordinate of first point
+   * @param y1 Relative Y co-ordinate of second point
+   * @param x2 Relative X co-ordinate of first point
+   * @param y2 Relative Y co-ordinate of first point
+   *
+   * @returns float
+   */
+  float getLineAngle(float x1, float y1, float x2, float y2)
+
+  /*
+   * @brief Converts `PostProcesser::ProcessedResults` to a `Pose::Pose`
+   *
+   * @param results  The PostProcessor::ProccessedResults struct from PostProcessor::PostProcessor being run
+   * @return JointsPose
+   */
+  Pose createPose(PostProcessing :: ProcessedResults results);
+
+  /*
+   * @brief sets user's current pose
+   *
+   * @param Pose
+   */
+  void set_current_pose(Pose current);
+
+  /*
+   * @brief sets user's ideal pose
+   *
+   * @param Pose
+   */
+  void set_ideal_pose(Pose current);
+
+  /*
+   * @brief sets calcaluted changes needed pose
+   *
+   * @param Pose
+   */
+  void set_pose_changes_needed(Pose current);
+
+ public:
+  /*
+   * @brief The user's current pose from most recent data provided
+   */
+  Pose current_pose;
+
+  /* 
+   * @brief The user's ideal pose, expected to be generated during user calibration  
+   */
+  Pose ideal_pose;
+
+  /*
+   * @brief A representation of what changes are needed to get back to ideal pose
+   */
+  Pose pose_changes_needed;
+
+  /*
+   * @brief A representation of what absolute value changes for a pose is acceptably still a good pose
+   */
+  float pose_change_threshold;
+
+  /*
+   * @brief Whether the user is in a good pose (true) or bad pose (false)
+   */
+  bool good_pose;
+
+  /*
+   * @brief Returns the user's current pose
+   * 
+   * @returns Pose
+   */
+  Pose get_current_pose();
```

Comment:
Would this be any different than accessing the `current_pose` field?

Reply:
That's the thing, the same way in Java you can get class.field, it's just standard practice to use getters and setters no?

Reply:
I'm happy to remove them. Just thought it was standard practice.